lib/helpers.js

```diff
@@ -849,20 +849,10 @@ function toAvdLocaleArgs (language, country) {
  */
 async function getAndroidPrefsRoot () {
   // https://gerrit.pixelexperience.org/plugins/gitiles/development/+/5c11852110eeb03dc5a69111354b383f98d15336/tools/androidprefs/src/com/android/prefs/AndroidLocation.java
-  let location = null;
-  if (system.isWindows()) {
-    let localAppData = process.env.LOCALAPPDATA;
-    if (!localAppData) {
-      localAppData = process.env.USERPROFILE;
-      if (localAppData) {
-        localAppData = path.resolve(localAppData, 'Local Settings', 'Application Data');
-      }
-    }
-    if (localAppData) {
-      location = path.resolve(localAppData, 'Android');
-    }
-  } else {
-    const home = process.env.HOME;
+  let location = process.env.ANDROID_EMULATOR_HOME;
+
+  if (!location) {
+    const home = process.env.ANDROID_SDK_HOME || process.env.HOME || process.env.USERPROFILE;
```

Comment:
ANDROID_AVD_HOME

Reply:
also it makes sense to not only check it is present but also that the actual directory exists

Reply:
same for ANDROID_EMULATOR_HOME 
I would also add some logging in case these env variables do not point to valid locations

Reply:
Thanks for your comments! The directory-exists-check is already done on line 861:
```js
if (location && await fs.exists(location))
```

> ANDROID_AVD_HOME/.. instead of ANDROID_SDK_HOME

I think it should be `ANDROID_SDK_HOME`, because this function returns the `.android` root folder, not the `avd` folder. The `listEmulators` function resolves the `avd` folder.

Reply:
> the directory-exists-check is already done on line 861

the is no such check for ANDROID_EMULATOR_HOME

Reply:
the documentation says

> ANDROID_AVD_HOME   Sets the path to the directory that contains all AVD-specific files, which mostly consist of very large disk images. The default location is $ANDROID_EMULATOR_HOME/avd/. You might want to specify a new location if the default location is low on disk space. 

And on my system:

```
$ echo $ANDROID_SDK_HOME                                                                                                                                                        
/Users/user/Library/Android/sdk
```

but the .android folder is located at `/Users/user`

Reply:
> the is no such check for ANDROID_EMULATOR_HOME

I've added the check

Reply:
> the documentation says
> 
> > ANDROID_AVD_HOME   Sets the path to the directory that contains all AVD-specific files, which mostly consist of very large disk images. The default location is $ANDROID_EMULATOR_HOME/avd/. You might want to specify a new location if the default location is low on disk space.
> 
> And on my system:
> 
> ```
> $ echo $ANDROID_SDK_HOME                                                                                                                                                        
> /Users/user/Library/Android/sdk
> ```
> 
> but the .android folder is located at `/Users/user`

`listEmulators()` appends the `avd` folder to the path returned by `getAndroidPrefsRoot()` (ln 77):
```js
avdsRoot = path.resolve(prefsRoot, 'avd');
```

So the return value of `getAndroidPrefsRoot()` should be the parent of `/avd`. `$ANDROID_EMULATOR_HOME` is that, according to the docs you referenced.

Reply:
`const home = process.env.ANDROID_SDK_HOME || process.env.HOME || process.env.USERPROFILE;` -> `const home = process.env.HOME || process.env.USERPROFILE;`

Reply:
Ah okay, removed that 👍